lib/distribute.c

```diff
@@ -458,6 +458,8 @@ int group_distribute_list_create_helper(
 
 int group_distribute_list_destroy(struct nb_cb_destroy_args *args)
 {
+	if (args->event != NB_EV_APPLY)
```

Comment:
would this be a better fix:

```
diff --git a/lib/northbound.c b/lib/northbound.c
index 0bc79d0277..6158fb5ada 100644
--- a/lib/northbound.c
+++ b/lib/northbound.c
@@ -1585,7 +1585,8 @@ static int nb_callback_destroy(struct nb_context *context,
        args.dnode = dnode;
        args.errmsg = errmsg;
        args.errmsg_len = errmsg_len;
-       ret = nb_node->cbs.destroy(&args);
+       if (args.event == NB_EV_APPLY)
+               ret = nb_node->cbs.destroy(&args);
 
        /* Detect and log unexpected errors. */
        switch (ret) {
```

I spot checked a few other destroy functions and they all had this check.